[depends] bump libnfs to 4.0.0

fuzzard opened the pull request:
## Description
Bumps libnfs to 4.0.0 

## Motivation and Context
We are running an old version 2.0.0. 4.0.0 brings a slew of fixes/updates including nfs4 support in the lib. This will require some core changes to potentially enable this however.

## How Has This Been Tested?
Build test for all depends platforms. Runtime tested macos.
Might be worth trying to get some community testing before merge.

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

fuzzard committed:
[depends] bump libnfs to 4.0.0

Rechi commented on tools/depends/target/libnfs/Makefile:
```diff
@@ -3,12 +3,14 @@ DEPS= ../../Makefile.include Makefile
 SOURCE=$(LIBNAME)-$(VERSION)
 ARCHIVE=$(SOURCE).tar.gz
 
+BASE_URL := https://github.com/sahlberg/libnfs/archive
```
This is wrong.

fuzzard replied:
Yeah, will update that, need to get the source tar on mirrors and drop this

AlwinEsch replied:
Hace it uploaded to our ftp and available now on http://mirrors.kodi.tv/build-deps/sources/libnfs-4.0.0.tar.gz

fuzzard replied:
Thanks Alwin, I'll fix this up today

howie-f commented:
fyi, testing this pr but had no luck connecting to nfsv4 from fire-tv. am i right in the assumption this should work then, or am i missing something?

MilhouseVH commented:
From [libnfs](https://github.com/sahlberg/libnfs)
```
NFSv4:
======
NFSv3 is the default but NFSv4 can be selected either by using the URL argument
version=4 or programatically calling nfs_set_version(nfs, NFS_V4) before
connecting to the server/share.
```
so it's unlikely Kodi supports NFSv4 without additional changes - maybe it's possible to pass `version=4` on the `nfs://` url?

MilhouseVH commented:
Try `nfs://server/path?version=4`

howie-f commented:
From libnfs: `nfs://<server|ipv4|ipv6>/path[?arg=val[&arg=val]*]`
>Try nfs://server/path?version=4

tried this with and without asterisk as well... no luck.
thx @MilhouseVH, not intended to start a support thread here ;)

AkariDN commented:
About Kodi current NFS implementation:
* It doesn't take into account URL options, so some code changes are required to pass ```version=4``` from url to libnfs.
* It uses deprecated ```nfs_stat``` function. We should use ```nfs_stat64``` with new library (in my case old function doesn't work at all).
* It always retrieves exports from ```mountd``` (like ```showmount -e```), so we can't use NFS4 without ```rpcbind``` and public ```mountd```. (```rpcbind``` is not required for NFS4 and ```mountd``` can be started locally - only one port is used). There should be an option to avoid using ```mountd```.

Moreover, it seems that Kodi is not well prepared for URLs with options. I've found at least two design bugs:
* Video database stores pathes without options (see ```CVideoDatabase::SplitPath```) but loads with options (see ```CVideoDatabase::GetPathId```). Watched state and other stuff will not work for such URLs without appropriate code changes.
* Options in URL can be unexpectedly reordered, but some comparison functions don't take it into accout. See ```CGUIDialogContextMenu::GetShare``` for example. Context menu on shares with options may not work properly. (Share created with ```version=4&norpc``` options may unexpectedly become ```norpc&version=4```)

Here are fast code changes to make NFS4 work (tested only on my system) with bugfixes:
https://github.com/AkariDN/xbmc/commit/24f6a8875823229bec22d0551bb2771c46d5b32d
URLs like ```nfs://myserver/myexport/?version=4&norpc``` can be used for shares, where all options are passed to libnfs (```version=4``` for NFS4) and ```norpc``` is a special option to avoid calls to ```mountd```.
You are free to use this code as you wish.

howie-f commented:
not sure at all - maybe this interferes/should be taken care of in vfs.nfs ??

fuzzard commented:
There's no intention to continue down the path of VFS.nfs/SMB at this point in time. From what I'm told, difficulties around add-on startup/loading and ordering of other things that require FS calls.

Commit:
[libnfs] bump to untagged commit to support PR #15686

Bump version to untagged commit hash to support features implemented in
https://github.com/xbmc/xbmc/pull/15686

fuzzard commented:
@peak3d have added a commit that bumps libnfs to a hash commited Jun 9th 2020 to support your timeout PR #15686

Builds ok on darwin_embedded, Maybe you want to try on android with your timeout PR if you find time?

fuzzard commented:
Jenkins build this please

peak3d commented:
@fuzzard works as expected on Android.

fuzzard commented:
Jenkins build this please
